include/fmt/chrono.h

```diff
@@ -1056,31 +1056,91 @@ void write_fractional_seconds(OutputIt& out, Duration d) {
   }
 }
 
+template <typename Char, typename OutputIt, typename Duration>
+void write_fractional_seconds(OutputIt& out, Duration d, int precision) {
+  constexpr auto num_fractional_digits =
+      count_fractional_digits<Duration::period::num,
+                              Duration::period::den>::value;
+
+  using subsecond_precision = std::chrono::duration<
+      typename std::common_type<typename Duration::rep,
+                                std::chrono::seconds::rep>::type,
+      std::ratio<1, detail::pow10(num_fractional_digits)>>;
+  if (precision > 0) {
+    *out++ = '.';
+    auto fractional =
+        detail::abs(d) - std::chrono::duration_cast<std::chrono::seconds>(d);
+    auto subseconds =
+        std::chrono::treat_as_floating_point<
+            typename subsecond_precision::rep>::value
+            ? fractional.count()
+            : std::chrono::duration_cast<subsecond_precision>(fractional)
+                  .count();
+    uint32_or_64_or_128_t<long long> n =
+        to_unsigned(to_nonnegative_int(subseconds, max_value<long long>()));
```

Comment:
Can subseconds be negative?

Reply:
It's nonnegative because `fractional` is nonnegative:
```const auto fractional =
      detail::abs(d) - std::chrono::duration_cast<std::chrono::seconds>(d);```

But subseconds can be signed or unsigned and `to_unsigned` raises error if argument is unsigned, so i'll make `static_cast`.